src/transformers/trainer.py

```diff
@@ -631,7 +631,7 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
         logger.info("***** Running training *****")
         logger.info("  Num examples = %d", self.num_examples(train_dataloader))
         logger.info("  Num Epochs = %d", num_train_epochs)
-        logger.info("  Instantaneous batch size per device = %d", self.args.per_device_train_batch_size)
+        logger.info("  Instantaneous batch size per device = %d", train_dataloader.batch_size)
```

Comment:
Why is this change necessary? The prompt is `Instantaneous batch size per device`. Also, this change seems to create inconsistency for logging.

Reply:
1. If using `--per_gpu_train_batch_size` (legacy flag for `per_device_train_batch_size`), the argument is straight up incorrect.
2. That batch size is per device.

Reply:
No the batch size is not per device. It is on TPUs but not if you a user has a `DataParallel` model.